### PR TITLE
Remove jQuery `.attr` from the common issue page functions

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched from jQuery `attr` to plain javascript `getAttribute` and `setAttribute`
- Tested most of the functions and they work as before

### CODE REVIEW

File: web_src/js/features/repo-issue.js [hunk 1]
```diff
@@ -123,9 +123,12 @@ export function initRepoIssueSidebarList() {
       fullTextSearch: true,
     });
 
-  function excludeLabel(item) {
-    const href = $(item).attr('href');
-    const id = $(item).data('label-id');
+  /**
+   * @param {HTMLElement} item
+   */
+  function excludeLabel(item) { // eslint-disable-line unicorn/consistent-function-scoping
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]: Move the function to top-level and remove the comment :)

File: web_src/js/features/repo-issue.js [hunk 2]
```diff
@@ -166,11 +169,11 @@ export function initRepoIssueCommentDelete() {
...
         if ($conversationHolder.find('.pending-label').length) {
-          const $counter = $('#review-box .review-comments-counter');
-          let num = parseInt($counter.attr('data-pending-comment-number')) - 1 || 0;
+          const counter = document.querySelector('#review-box .review-comments-counter');
+          let num = parseInt(counter?.getAttribute('data-pending-comment-number')) - 1 || 0;
           num = Math.max(num, 0);
-          $counter.attr('data-pending-comment-number', num);
-          $counter.text(num);
+          counter.setAttribute('data-pending-comment-number', num);
+          counter.textContent = num;
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]:
```suggestion
          counter.textContent = String(num);
```

File: web_src/js/features/repo-issue.js [hunk 3]
```diff
@@ -375,17 +382,17 @@ export function initRepoIssueComments() {
     window.location.reload();
   });
 
-  $(document).on('click', (event) => {
-    const $urlTarget = $(':target');
-    if (!$urlTarget.length) return;
+  document.addEventListener('click', (event) => {
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]:
```suggestion
  document.addEventListener('click', (e) => {
```
combine with suggestion below.

File: web_src/js/features/repo-issue.js [hunk 4]
```diff
@@ -375,17 +382,17 @@ export function initRepoIssueComments() {
...
+    const urlTargetId = urlTarget.id;
     if (!urlTargetId) return;
-    if (!/^(issue|pull)(comment)?-\d+$/.test(urlTargetId)) return;
 
-    const $target = $(event.target);
+    if (!/^(issue|pull)(comment)?-\d+$/.test(urlTargetId)) return;
 
-    if (!$target.closest(`#${urlTargetId}`).length) {
+    const target = event.target;
+    if (!target.closest(`#${urlTargetId}`)) {
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]:
```suggestion
    if (!e.target.closest(`#${urlTargetId}`)) {
```
combine with suggestion above.

File: web_src/js/features/repo-issue.js [hunk 5]
```diff
@@ -607,7 +617,7 @@ export function initRepoIssueTitleEdit() {
   $('#edit-title').on('click', editTitleToggle);
   $('#cancel-edit-title').on('click', editTitleToggle);
   $('#save-edit-title').on('click', editTitleToggle).on('click', async function () {
-    const pullrequest_targetbranch_change = async function (update_url) {
+    const pullrequest_targetbranch_change = async function (update_url) {// eslint-disable-line unicorn/consistent-function-scoping
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]: Also move. Maybe as `async function pullRequestTargetBranchChange(update_url)` top-level function.

File: web_src/js/features/repo-issue.js [hunk 6]
```diff
@@ -671,10 +681,11 @@ export function initSingleCommentEditor($commentForm) {
   // * normal new issue/pr page, no status-button
   // * issue/pr view page, with comment form, has status-button
   const opts = {};
-  const $statusButton = $('#status-button');
-  if ($statusButton.length) {
+  const statusButton = document.getElementById('status-button');
+  if (statusButton) {
     opts.onContentChanged = (editor) => {
-      $statusButton.text($statusButton.attr(editor.value().trim() ? 'data-status-and-comment' : 'data-status'));
+      const statusText = editor.value().trim() ? statusButton.getAttribute('data-status-and-comment') : statusButton.getAttribute('data-status');
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]:
```suggestion
      const statusText = statusButton.getAttribute(editor.value().trim() ? 'data-status-and-comment' : 'data-status');
```

Shorter this way.

File: web_src/js/features/repo-issue.js [hunk 7]
```diff
@@ -144,9 +147,9 @@ export function initRepoIssueSidebarList() {
 
   $('.menu .ui.dropdown.label-filter').on('keydown', (e) => {
     if (e.altKey && e.keyCode === 13) {
-      const $selectedItems = $('.menu .ui.dropdown.label-filter .menu .item.selected');
-      if ($selectedItems.length > 0) {
-        excludeLabel($($selectedItems[0]));
+      const selectedItems = document.querySelectorAll('.menu .ui.dropdown.label-filter .menu .item.selected');
+      if (selectedItems.length > 0) {
+        excludeLabel(selectedItems[0]);
```
Comment (delvh) [inline web_src/js/features/repo-issue.js]: That is weird code
```suggestion
      const selectedItem = document.querySelector('.menu .ui.dropdown.label-filter .menu .item.selected');
      if (selectedItem) {
        excludeLabel(selectedItem);
```

File: web_src/js/features/repo-issue.js (hunk 1 above)
Comment (yardenshoham) [reply web_src/js/features/repo-issue.js]: [`1579b43` (#30083)](https://github.com/go-gitea/gitea/pull/30083/commits/1579b43671b6b1bddf53187c9c8ca6f837d63c51)

File: web_src/js/features/repo-issue.js (hunk 5 above)
Comment (yardenshoham) [reply web_src/js/features/repo-issue.js]: [`1579b43` (#30083)](https://github.com/go-gitea/gitea/pull/30083/commits/1579b43671b6b1bddf53187c9c8ca6f837d63c51)

**Review by silverwind — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**